go/database/mpt/forest_test.go

```diff
@@ -2001,14 +2048,43 @@ func TestForest_FailingFlush_Invalidates_Forest(t *testing.T) {
 		t.Fatalf("failed to create test forest: %v", err)
 	}
 
-	// only call fails, but the forest must be invalidated
+	defer func() {
+		if err = forest.Close(); err == nil {
+			t.Fatalf("close should fail")
+		}
+		// Error returned by first flush is kept up to this point
+		if !errors.Is(err, injectedError) {
```

Comment:
I think it covers also situation when `err == nil`, i.e. can be merged with condition above